src/lib/progress-bar/progress-bar.ts

```diff
@@ -113,6 +131,21 @@ export class MatProgressBar extends _MatProgressBarMixinBase implements CanColor
       return {transform: `scaleX(${scale})`};
     }
   }
+
+  /** Attaches value bar element animation end listener. */
+  ngAfterViewInit() {
+    if(this.mode === 'determinate' || this.mode === 'buffer') {
```

Comment:
Rather than only observing for animation events when it is `determinate` or `buffer`, we should always set up the subscription and only emit out `valueAnimationEnd` event when appropriate.  Otherwise, the `mode` of the `MatProgressBar` cannot be changed dynamically and still utilize this event

Reply:
Oh that's nice tip! Definitely more flexible that way, I guess one possible downside is that the output is always executed on unsupported method, but that seems like small overhead?